app/containers/BookmarkList/index.js

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * BookmarkList
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Helmet from 'react-helmet';
+import { List, fromJS } from 'immutable';
+
+import { loadEntitiesIfNeeded, updatePath } from 'containers/App/actions';
+import { selectReady } from 'containers/App/selectors';
+import appMessages from 'containers/App/messages';
+import { PATHS } from 'containers/App/constants';
+
+import EntityList from 'containers/EntityList';
+
+import { CONFIG, DEPENDENCIES } from './constants';
+import { selectBookmarks } from './selectors';
+import messages from './messages';
+
+export class BookmarkList extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
+  componentWillMount() {
+    this.props.loadEntitiesIfNeeded();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // reload entities if invalidated
+    if (!nextProps.dataReady) {
+      this.props.loadEntitiesIfNeeded();
+    }
+  }
+
+  render() {
+    const { dataReady } = this.props;
+
+    const headerOptions = {
+      supTitle: this.context.intl.formatMessage(messages.pageTitle),
+      icon: 'pages',
+      actions: [],
+    };
+
+    return (
+      <div>
+        <Helmet
+          title={this.context.intl.formatMessage(messages.pageTitle)}
+          meta={[
+            { name: 'description', content: this.context.intl.formatMessage(messages.metaDescription) },
+          ]}
+        />
+        <EntityList
+          entities={this.props.entities}
+          config={CONFIG}
+          header={headerOptions}
+          dataReady={dataReady}
+          entityTitle={{
+            single: this.context.intl.formatMessage(appMessages.entities.bookmarks.single),
+            plural: this.context.intl.formatMessage(appMessages.entities.bookmarks.plural),
+          }}
+          locationQuery={fromJS(this.props.location.query)}
+        />
```

Comment:
I would probably pass a special "entity click handler" as a prop to the entity list that it would use [here](https://github.com/impactoss/impactoss-client/blob/bookmark-custom-views/app/containers/EntityList/index.js#L188) if present.

then this list view could trigger an action (with the json object as payload) that would be handled by a saga that would update the path to resurrect the view

Reply:
Hey @tmfrnz, please check the code again. Entities was a list, so I had to find the right one first with the sent it. Then with the utils function that converted the view to a url, I used the existing updatePath saga.